tests/src/resultcollection.hh

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: 2022 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+#include "testcommon.hh"
+
+#include <ikarus/finiteelements/ferequirements.hh>
+
+auto createLinearStressResults = []<typename NonLinearOperator, typename FiniteElement>(
+                                     NonLinearOperator& nonLinearOperator, FiniteElement& fe) {
+  constexpr int dim = FiniteElement::Traits::mydim;
+
+  static_assert(std::remove_cvref_t<decltype(fe.localView().tree())>::degree() == dim,
+                "The test to check linear stress is only supported with the powerBasis having the same power as the "
+                "element dimension");
+  static_assert(
+      std::is_same_v<
+          std::remove_cvref_t<decltype(fe.localView().tree().child(0))>,
+          Dune::Functions::LagrangeNode<std::remove_cvref_t<decltype(fe.localView().globalBasis().gridView())>, 1>>,
+      "The test to check linear stress is only supported for a linear Lagrange basis");
+
+  const auto& element          = fe.gridElement();
+  const auto& referenceElement = Dune::referenceElement<double, dim>(element.type());
+  const int nodes              = referenceElement.size(dim);
+
+  Eigen::MatrixXd expectedStress(nodes, dim * (dim + 1) / 2);
+
```

Comment:
Okay so you want for each Elementshape + dimension a different function that fives the expectedResult? What about EAS?

- [ ] Q1 2D
- [ ] Q1 3D
- [ ] Q1 2D EAS
- [ ] T1 2D
- [ ] T1 3D

Reply:
Yes so basically seperating `createLinearStressResults` for each test case as stated above.
For EAS i think we still have to pass the fe, since we have to call setEASType. which is unfortunate :/. But I don't see way around it currently.

Reply:
Okay, i would prefer the functions to all have the same interface, so all taking the fe argument, what do you think?
E.g.

```cpp
auto createLinearStressResultsForSimplex = []<typename FiniteElement>(FiniteElement& fe) {}
```
or as we don't really need the type this would be enough
```cpp
auto createLinearStressResultsForSimplex = [](auto& fe) {}
```

Reply:
yes i think all should accept the fe argument

Reply:
Also i need the nonlinop as of now, because when I do that
```cpp
Eigen::VectorXd displacement;
displacement << 0, 0, 2, 0, 1, 0;
.insertGlobalSolution(FESolutions::displacement, displacement)
```
the `displacment` goes out of scope and is not longer available in the test function. Thats why earlier iteration of this were doing 
```cpp
  auto& displacement = nonLinearOperator.firstParameter();
```

Reply:
You can overload https://github.com/ikarus-project/ikarus/blob/7e793d0ae955b4b11ba7d2c98522775495abe703/ikarus/finiteelements/mechanics/linearelastic.hh#L32

and the same way here: https://github.com/ikarus-project/ikarus/blob/7e793d0ae955b4b11ba7d2c98522775495abe703/ikarus/finiteelements/ferequirements.hh#L117

such that `SolutionVectorType_ ` is `Eigen::VectorXd` instead of a reference to it.

But since also passing the nonLinearOperator anyway I'm fine with the current solution, but anyway it would be nice anyway. But I'm not sure if this would be worth it.

Reply:
Yeah i think its not too bad to have the nonLinOp as a function argument as its already available in these test functors